Contiguous.Rmd

```diff
@@ -0,0 +1,298 @@
+
+```{python}
+assert np.allclose(bike_thefts, b_sun * sunshine + c_sun + errors_sun)
+```
+
+This test has passed, suggesting that the values have been calculated properly.
+
+```{python}
+scatter_error(sunshine, bike_thefts, fitted_sun)
+```
+
+It seems that there is a stronger predictive relationship present between sunshine hours and bike thefts, than rainfall and bike thefts. Based on the error intervals, it seems that this predictive relationship holds strongest at the extremes; when it is sunniest and darkest.
+
+The relationship is a bit weaker towards the middle values, suggesting that there may be a confounding variable that is responsible for these results.
+
+The clustering follows the BST and GMT groupings, as expected with sunshine hours, with the months in which the change occurs (Oct and Mar) being close together. What is interesting as that despite the fact that Mar and Oct have a very similar number of daylight hours, there is a very big difference in the actual values. This could be the effect of the time zone change.
+
+The end of BST is at the end of Oct, which means for the most of month society has an extra hour of daylight, even though there are less sunshine hours than Mar. The end of GMT is at the end of Mar, which means that there is one less hour of daylight, even though there are more sunshine hours than Oct. It could be that the stark difference between the actual values for the two months are a result of the change in time zones.
+
+
+### Temperature
+Temperature is also a factor that changes as the time zones do, while being a major influence on people's behaviour. People might be more inclined to cycle in BST, as it is warmer.
+
+```{python}
+max_temps = np.array([8.42, 8.98, 11.73, 15, 18.37, 21.57, 23.89, 23.40, 20.22, 15.81, 11.47, 8.79])
+min_temps = np.array([2.68, 2.65, 4.14, 6.03, 9.08, 12.03, 14.18, 14.06, 11.62, 8.78, 5.26, 3.08])
+avg_temps = (max_temps + min_temps) / 2
+temp_dict = {'Months':['Jan', 'Feb', 'Mar', 'Apr', 'May', 'Jun', 'Jul', 'Aug', 'Sep', 'Oct', 'Nov', 'Dec'], 'Avg Temperature (°C)': avg_temps}
+temp_df = pd.DataFrame(data=temp_dict)             
+temp_df
+```
+
+The dataframe above shows the average temperature in London per month using data collected from 1991-2020 by the [Met Office](https://metoffice.gov.uk/research/climate/maps-and-data/uk-climate-averages/gcpsvg3nc). The data provided by the Met Office was the minimum and maximum temperatures per month. For the purposes of this analysis, I calculated the mean temperature per month using the two values provided.
+
+```{python}
+temp_df.plot()
+plt.ylabel('Temperature')
+plt.xlabel('Month')
+```
+
+```{python}
+theft_df.plot()
+plt.ylabel('Bike Thefts')
+```
+
+Like with the sunshine hours, there seems to be a high degree of similarity between the two plots of bike thefts per month and average temparature per month. This could be suggestive of a strong predictive relationship between the two variables.
```

Comment:
```suggestion
Like with the sunshine hours, there seems to be a high degree of similarity between the two plots of bike thefts per month and average temperature per month. This could be suggestive of a strong predictive relationship between the two variables.
```